docs/source/training.rst

```diff
@@ -10,290 +10,362 @@
+    tf_eval_dataset = small_eval_dataset.remove_columns(["text"]).with_format("tensorflow")
 
-Let's use ``tensorflow_datasets`` to load in the `MRPC dataset
-<https://www.tensorflow.org/datasets/catalog/glue#gluemrpc>`_ from GLUE. We can then use our built-in
-:func:`~transformers.data.processors.glue.glue_convert_examples_to_features` to tokenize MRPC and convert it to a
-TensorFlow ``Dataset`` object. Note that tokenizers are framework-agnostic, so there is no need to prepend ``TF`` to
-the pretrained tokenizer name.
+Then we convert everything in big tensors and use the :obj:`tf.data.Dataset.from_tensor_slices` method:
 
 .. code-block:: python
 
-    from transformers import BertTokenizer, glue_convert_examples_to_features
-    import tensorflow as tf
-    import tensorflow_datasets as tfds
-    tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
-    data = tfds.load('glue/mrpc')
-    train_dataset = glue_convert_examples_to_features(data['train'], tokenizer, max_length=128, task='mrpc')
-    train_dataset = train_dataset.shuffle(100).batch(32).repeat(2)
+    train_features = {x: tf_train_dataset[x].to_tensor() for x in tokenizer.model_input_names}
+    train_tf_dataset = tf.data.Dataset.from_tensor_slices((train_features, tf_train_dataset["label"]))
+    train_tf_dataset = train_tf_dataset.shuffle(len(tf_train_dataset)).batch(8)
+
+    eval_features = {x: tf_eval_dataset[x].to_tensor() for x in tokenizer.model_input_names}
+    eval_tf_dataset = tf.data.Dataset.from_tensor_slices((eval_features, tf_eval_dataset["label"]))
+    eval_tf_dataset = eval_tf_dataset.batch(8)
 
-The model can then be compiled and trained as any Keras model:
+With this done, the model can then be compiled and trained as any Keras model:
 
 .. code-block:: python
 
-    optimizer = tf.keras.optimizers.Adam(learning_rate=3e-5)
-    loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
-    model.compile(optimizer=optimizer, loss=loss)
-    model.fit(train_dataset, epochs=2, steps_per_epoch=115)
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(learning_rate=5e-5),
+        loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        metrics=tf.metrics.SparseCategoricalAccuracy(),
+    )
+
+    model.fit(train_tf_dataset, validation_data=eval_tf_dataset, epochs=3)
 
 With the tight interoperability between TensorFlow and PyTorch models, you can even save the model and then reload it
 as a PyTorch model (or vice-versa):
 
 .. code-block:: python
 
-    from transformers import BertForSequenceClassification
-    model.save_pretrained('./my_mrpc_model/')
-    pytorch_model = BertForSequenceClassification.from_pretrained('./my_mrpc_model/', from_tf=True)
+    from transformers import AutoModelForSequenceClassification
 
+    model.save_pretrained("my_imdb_model")
+    pytorch_model = TFAutoModelForSequenceClassification.from_pretrained("my_imdb_model", from_tf=True)
```

Comment:
```suggestion
    pytorch_model = AutoModelForSequenceClassification.from_pretrained("my_imdb_model", from_tf=True)
```

You accidentally used the wrong Class here

Reply:
By the way when running this I got the following warning:

```
All TF 2.0 model weights were used when initializing BertForSequenceClassification.

Some weights of BertForSequenceClassification were not initialized from the TF 2.0 model and are newly initialized: ['bert.embeddings.position_ids']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
```

This is odd as we just trained/fine-tuned this model (albeit with TensorFlow), so we shouldn't be getting this error at all?